src/synchronizeContacts.js

```diff
@@ -2,9 +2,61 @@ const get = require('lodash/get')
 const pLimit = require('p-limit')
 const mergeWith = require('lodash/mergeWith')
 const isArray = require('lodash/isArray')
+const isEqual = require('lodash/isEqual')
 const unset = require('lodash/unset')
+const omit = require('lodash/omit')
 const transpileContactToCozy = require('./helpers/transpileContactToCozy')
 
+const customMerge = connectorGroupIds => (existingValue, newValue, key) => {
+  if (key === 'cozy') return cozyMergeStrategy(existingValue, newValue)
+  if (key === 'relationships')
+    return relationshipsMergeStrategy(connectorGroupIds)(
+      existingValue,
+      newValue
+    )
+  else return undefined // for other fields, use the normal lodash merge strategy
+}
+
+// for the cozy field, we want to always the new value
+const cozyMergeStrategy = (existingCozy, newCozy) => {
+  const hasNewCozyValue = isArray(newCozy) && newCozy.length > 0
+  if (hasNewCozyValue) return newCozy
+  else return undefined
+}
+
+const relationshipsMergeStrategy = connectorGroupIds => (
+  existingRelationships,
+  newRelationships
+) => {
+  const otherRelationships = omit(existingRelationships, 'groups')
+  const existingGroups = get(existingRelationships, 'groups.data', [])
+  const existingManualGroups = existingGroups.filter(
+    ({ _id: existingGroupId }) =>
+      !connectorGroupIds.some(
+        ({ _id: connectorGroupId }) => connectorGroupId === existingGroupId
+      )
+  )
+  const newGroups = get(newRelationships, 'groups.data', [])
+
+  return {
+    ...otherRelationships,
+    groups: {
+      data: [...newGroups, ...existingManualGroups]
+    }
+  }
+}
+
+const getFinalContactData = (cozyContact, remoteContact, connectorGroupIds) => {
+  const merged = mergeWith(
+    {},
```

Comment:
maybe add a comment here to explain that this argument is here to avoid mutation of `cozyContact` (if I understood it well)